packages/mui-joy/src/Badge/Badge.tsx

```diff
@@ -0,0 +1,326 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import BadgeUnstyled from '@mui/base/BadgeUnstyled';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize, usePreviousProps } from '@mui/utils';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
+import shouldSpreadAdditionalProps from '../utils/shouldSpreadAdditionalProps';
+import badgeClasses, { getBadgeUtilityClass } from './badgeClasses';
+import { BadgeProps, BadgeTypeMap } from './BadgeProps';
+
+const useUtilityClasses = (ownerState: BadgeProps) => {
+  const { color, variant, size, anchorOrigin, invisible } = ownerState;
+
+  const slots = {
+    root: ['root'],
+    badge: [
+      'badge',
+      invisible && 'invisible',
+      anchorOrigin &&
+        `anchorOrigin${capitalize(anchorOrigin.vertical)}${capitalize(anchorOrigin.horizontal)}`,
+      variant && `variant${capitalize(variant)}`,
+      color && `color${capitalize(color)}`,
+      size && `size${capitalize(size)}`,
+    ],
+  };
+
+  return composeClasses(slots, getBadgeUtilityClass, {});
+};
+
+const BadgeRoot = styled('span', {
+  name: 'MuiBadge',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: BadgeProps }>(() => ({
+  position: 'relative',
+  display: 'inline-flex',
+  // For correct alignment with the text.
+  verticalAlign: 'middle',
+  flexShrink: 0,
+}));
+
+const BadgeBadge = styled('span', {
+  name: 'MuiBadge',
+  slot: 'Badge',
+  overridesResolver: (props, styles) => styles.badge,
+})<{ ownerState: BadgeProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      ...(ownerState.size === 'sm' && {
+        '--Badge-minWidth': '1rem',
+        '--Badge-height': '1rem',
+        '--Badge-borderRadius': '0.5rem',
+        '--Badge-fontSize': theme.vars.fontSize.sm,
+      }),
+      ...(ownerState.size === 'md' && {
+        '--Badge-minWidth': '1.25rem',
+        '--Badge-height': '1.25rem',
+        '--Badge-borderRadius': '0.625rem',
+        '--Badge-fontSize': theme.vars.fontSize.md,
+      }),
+      ...(ownerState.size === 'lg' && {
+        '--Badge-minWidth': '1.5rem',
+        '--Badge-height': '1.5rem',
+        '--Badge-borderRadius': '0.75rem',
+        '--Badge-fontSize': theme.vars.fontSize.lg,
+      }),
+      display: 'flex',
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      justifyContent: 'center',
+      alignContent: 'center',
+      alignItems: 'center',
+      position: 'absolute',
+      boxSizing: 'border-box',
+      fontFamily: theme.vars.fontFamily.body,
+      fontWeight: theme.vars.fontWeight.sm,
```

Comment:
```suggestion
      fontWeight: theme.vars.fontWeight.md,
```
Would probably iterate with something thicker here...